core/modules/views/views.install

```diff
@@ -129,6 +128,28 @@ function views_update_1003() {
 }
 
 /**
+ * Replace the frontpage view with promoted.
+ */
+function views_update_1004() {
+  // If the old frontpage view is still disabled, delete it.
+  if (config_get('views.view.frontpage', 'disabled') === TRUE) {
+    $config_frontpage = config('views.view.frontpage');
+    $config_frontpage->delete();
+  }
+  // See if there is already a promoted view on this site.
+  $config_promoted = config('views.view.promoted');
+  if (is_object($config_promoted) && $config_promoted->isNew()) {
```

Comment:
`is_object($config_promoted) && ` is unnecessary. `config()` will always return an object even if the config doesn't exist.